MobileBuy/buy/src/main/java/com/shopify/buy/utils/AndroidPayHelper.java

```diff
@@ -0,0 +1,394 @@
+            address2.append(walletUserAddress.getAddress4());
+            address2.append(", ");
+        }
+        if (!TextUtils.isEmpty(walletUserAddress.getAddress4())) {
+            address2.append(walletUserAddress.getAddress4());
+            address2.append(", ");
+        }
+        if (!TextUtils.isEmpty(walletUserAddress.getAddress5())) {
+            address2.append(walletUserAddress.getAddress5());
+        }
+
+        address.setAddress2(address2.toString());
+
+        address.setZip(walletUserAddress.getPostalCode());
+        address.setCity(walletUserAddress.getLocality());
+        address.setCountryCode(walletUserAddress.getCountryCode());
+        address.setProvinceCode(walletUserAddress.getAdministrativeArea());
+        address.setPhone(walletUserAddress.getPhoneNumber());
+        return address;
+    }
+
+    private static String[] extractFirstAndLastNames(String name) {
+        String[] firstAndLastName = new String[2];
+        String[] nameParts = name.split(" ");
+        if (nameParts.length > 0) {
+            firstAndLastName[FIRST_NAME_PART] = (nameParts[FIRST_NAME_PART]);
+        }
+        int nameIndex = 1;
+        String lastName = "";
+        while (nameParts.length > nameIndex) {
+            lastName += nameParts[nameIndex++];
+        }
+        firstAndLastName[LAST_NAME_PART] = lastName.trim();
+        return firstAndLastName;
+    }
+
+    /**
+     * Creates a Masked Wallet Request from a Shopify Checkout
+     *
+     * @param merchantName The merchant name to show on the Android Pay dialogs, not null or empty
+     * @param checkout     The {@link Checkout} to use, not null.
+     * @param buyClient    The {@link BuyClient} to use, not null.  Must have previously had Android Pay enabled via {@link BuyClient#enableAndroidPay(String)}
+     * @param phoneNumberRequired If true, the phone number will be required as part of the Shipping Address in Android Pay
+     * @return A {@link MaskedWalletRequest}
+     */
+    public static MaskedWalletRequest createMaskedWalletRequest(String merchantName, Checkout checkout, BuyClient buyClient, boolean phoneNumberRequired) {
+        if (checkout == null) {
+            throw new NullPointerException("checkout cannot be null");
+        }
+
+        if (buyClient == null) {
+            throw new NullPointerException("buyClient cannot be null");
+        }
+
+        if (TextUtils.isEmpty(merchantName)) {
+            throw new IllegalArgumentException("merchantName cannot be empty");
+        }
+
+        // Create the parameters that will be used for encrypting Network Tokens.
+        PaymentMethodTokenizationParameters parameters =
+                PaymentMethodTokenizationParameters.newBuilder()
+                        .setPaymentMethodTokenizationType(PaymentMethodTokenizationType.NETWORK_TOKEN)
+                        .addParameter("publicKey", buyClient.getAndroidPayPublicKey())
```

Comment:
What if Android Pay is disabled, in this case pay public key is null, we should add verification if pay is not disabled for this function